data/state/journalEntries.go

```diff
@@ -1,34 +1,120 @@
 package state
 
 import (
+	"bytes"
 	"fmt"
 
 	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/marshal"
 )
 
 type journalEntryCode struct {
-	codeHash []byte
-	updater  Updater
+	oldCodeEntry    *CodeEntry
+	oldCodeHash     []byte
+	newCodeHash     []byte
+	codeForEviction map[string]struct{}
+	newCode         map[string]struct{}
+	trie            Updater
+	marshalizer     marshal.Marshalizer
 }
 
 // NewJournalEntryCode creates a new instance of JournalEntryCode
-func NewJournalEntryCode(codeHash []byte, updater Updater) (*journalEntryCode, error) {
-	if check.IfNil(updater) {
+func NewJournalEntryCode(
+	oldCodeEntry *CodeEntry,
+	oldCodeHash []byte,
+	newCodeHash []byte,
+	codeForEviction map[string]struct{},
+	newCode map[string]struct{},
+	trie Updater,
+	marshalizer marshal.Marshalizer,
+) (*journalEntryCode, error) {
+	if check.IfNil(trie) {
 		return nil, ErrNilUpdater
 	}
-	if len(codeHash) == 0 {
-		return nil, ErrInvalidHash
+	if check.IfNil(marshalizer) {
+		return nil, ErrNilMarshalizer
+	}
+	if codeForEviction == nil {
+		return nil, ErrNilCodeForEvictionMap
+	}
+	if newCode == nil {
+		return nil, ErrNilNewCodeMap
 	}
 
 	return &journalEntryCode{
-		codeHash: codeHash,
-		updater:  updater,
+		oldCodeEntry:    oldCodeEntry,
```

Comment:
oldCodeEntry could be nil? If not you should check it above.

Reply:
oldCodeEntry could be nil in the following case: If a newCode is set to an account that had no code before, the oldCodeEntry should be nil, in case that a revert occurs.

Reply:
oldCodeEntry could be nil in the following case: If a newCode is set to an account that had no code before, the oldCodeEntry should be nil, in case that a revert occurs.

Reply:
ok